code/script.js

```diff
@@ -262,82 +417,163 @@ const selectQuestion = () => {
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  const keep = currentQuestion.value === secret[currentQuestion.attribute]
+  filterCharacters(keep)
+ // Compare the currentQuestion with the secret person.
+// See if we should keep or remove people based on that
+// Then invoke filterCharacters
 }
-
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
-      )
-    }
-  } else if (group === 'other') {
-    // Similar to the one above
+// Show the correct alert message for different categories
+const { attribute, category, value } = currentQuestion
+
+if (category === 'accessories') {
+  if (keep) {
+    alert(
+      `You are correct, the person wears ${attribute}! Keep all that wears ${attribute}.`
+    )
   } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
-    } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
-    }
+    alert(
+      `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}.`
+    )
   }
-
+} else if (category === 'hair color') {
+  if (keep) {
+    alert(`Sure thing, the person has ${value} hair! Keep all that has ${value} hair.`
+    )
+  } else {
+    alert(
+    `Nope, the person doesn't have ${value} hair. Remove all that has ${value} hair.`
+    )
+  }
+} else if (category === 'eye color') {
+  if (keep) {
+    alert(
+      `Correctamundo, the person has ${value} eyes! Keep all that has ${value} eyes.`
+    )
+  } else {
+    alert(
+      `Sorry, the person doesn't have ${value} eyes. Remove all that has ${value} eyes.`
+      )
+  }
+} else if (category === 'other') {
+  if (keep) {
+    alert(
+      `Affirmative, the person has a ${attribute}. Keep all that has a ${attribute}.`
+    )
+} else {
+    alert(
+      `Negative, the person doesn't have a ${attribute}. Remove all that has a ${attribute}.`
+    )
+  } 
+}
   // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
+  
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
```

Comment:
Nice logic! 😻 Don't you just love it when complex stuff can be written in one line! 🥇